src/Umbraco.Core/Services/Implement/EntityService.cs

```diff
@@ -257,20 +258,38 @@ public virtual IEntitySlim GetParent(int id, UmbracoObjectTypes objectType)
         /// <inheritdoc />
         public virtual IEnumerable<IEntitySlim> GetChildren(int parentId)
         {
-            using (ScopeProvider.CreateScope(autoComplete: true))
+            using (var scope = ScopeProvider.CreateScope(autoComplete: true))
             {
-                var query = Query<IUmbracoEntity>().Where(x => x.ParentId == parentId);
-                return _entityRepository.GetByQuery(query);
+                try
+                {
+                    var query = Query<IUmbracoEntity>().Where(x => x.ParentId == parentId);
+                    return _entityRepository.GetByQuery(query);
+                }
+                catch (LockTimeoutException e)
```

Comment:
I guess this means that we are effectively going to have to wrap every call to any repository query with this?

Reply:
If we want a custom reason then yes, otherwize if falls back to the default